src/main.rs

```diff
@@ -179,5 +205,7 @@ mod tests {
 
         assert_eq!(oracle_message.validator_public_key, public_key);
         assert!(oracle_message.interval_inclusion_messages.len() > 100);
+
+        mock.assert();
```

Comment:
Just for my understanding, what does this assert / check exactly ?

Reply:
That the mock server was actually called and not maybe a real server instead.

Reply:
Yes exactly it does check that mock server was called